crates/loam-cli/src/commands/build/build_clients.rs

```diff
@@ -0,0 +1,260 @@
+            return Err(Error::NoSettingsForCurrentEnv);
+        };
+        let current_env = current_env.unwrap();
+
+        self.add_network_to_env(&current_env.network)?;
+        self.handle_accounts(&current_env.accounts).await?;
+        self.handle_contracts(&current_env.contracts).await?;
+
+        Ok(())
+    }
+
+    /// Parse the network settings from the environments.toml file and set STELLAR_RPC_URL and
+    /// STELLAR_NETWORK_PASSPHRASE.
+    ///
+    /// We could set STELLAR_NETWORK instead, but when importing contracts, we want to hard-code
+    /// the network passphrase. So if given a network name, we use soroban-cli to fetch the RPC url
+    /// & passphrase for that named network, and still set the environment variables.
+    fn add_network_to_env(&self, network: &Network) -> Result<(), Error> {
+        let rpc_url = &network.rpc_url;
+        let network_passphrase = &network.network_passphrase;
+        let network_name = &network.name;
+
+        if let Some(name) = network_name {
+            let cli::network::Network {
+                rpc_url,
+                network_passphrase,
+            } = (cli::network::Args {
+                network: Some(name.clone()),
+                rpc_url: None,
+                network_passphrase: None,
+            })
+            .get(&cli::config::locator::Args {
+                global: false,
+                config_dir: None,
+            })?;
+            println!("🌐 using {name} network");
+            std::env::set_var("STELLAR_RPC_URL", rpc_url);
+            std::env::set_var("STELLAR_NETWORK_PASSPHRASE", network_passphrase);
+        } else if let Some(rpc_url) = rpc_url {
+            if let Some(passphrase) = network_passphrase {
+                std::env::set_var("STELLAR_RPC_URL", rpc_url);
+                std::env::set_var("STELLAR_NETWORK_PASSPHRASE", passphrase);
+                println!("🌐 using network at {rpc_url}");
+            } else {
+                return Err(Error::MalformedNetwork);
+            }
+        }
+
+        Ok(())
+    }
+
+    async fn handle_accounts(&self, accounts: &Option<Vec<Account>>) -> Result<(), Error> {
+        if accounts.is_none() {
+            return Err(Error::NeedAtLeastOneAccount);
+        }
+
+        let accounts = accounts.as_ref().unwrap();
+
+        let default_account_candidates = accounts
+            .iter()
+            .filter_map(|account| {
+                account
+                    .default
+                    .unwrap_or(false)
+                    .then(|| account.name.clone())
+            })
+            .collect::<Vec<_>>();
+
+        let default_account = match default_account_candidates.len() {
+            0 if accounts.is_empty() => return Err(Error::NeedAtLeastOneAccount),
+            0 => accounts[0].name.clone(),
+            1 => default_account_candidates[0].to_string(),
+            _ => return Err(Error::OnlyOneDefaultAccount(default_account_candidates)),
+        };
```

Comment:
```suggestion
        let default_account = match (default_account_candidates, accounts) {
            ([], []) => return Err(Error::NeedAtLeastOneAccount),
            ([], [Account {name, ..}, ..] => name.clone(),
            ([candidate], _) => candidate.to_string(),
            _ => return Err(Error::OnlyOneDefaultAccount(default_account_candidates)),
        };
```

Reply:
Also do you need `candidate.to_string`

Reply:
Your suggestion doesn't work for a bunch of reasons and my Rust is not good enough to get it to work. Something to do with converting vecs to slices and then matching with arrays and then `the size for values of type `[Account]` cannot be known at compilation time`.

Reply:
Yes, we need to the `to_string` to work around 

```
cannot move out of index of `Vec<std::string::String>`
move occurs because value has type `std::string::String`, which does not
implement the `Copy` trait (rustc E0507)

Reply:
If you want to edit the code yourself and push up a change, I'd love to see how to do that nicer `match`